cast/src/starknet_commands/declare.rs

```diff
@@ -14,41 +13,110 @@ use starknet::{
     signers::LocalWallet,
 };
 use std::sync::Arc;
+use std::process::Command;
+use serde::Deserialize;
+use std::path::PathBuf;
+
+#[allow(dead_code)]
+#[derive(Deserialize)]
+struct ScarbStarknetArtifacts {
+    version: u32,
+    contracts: Vec<ScarbStarknetContract>,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize)]
+struct ScarbStarknetContract {
+    id: String,
+    package_name: String,
+    contract_name: String,
+    artifacts: ScarbStarknetContractArtifact,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize)]
+struct ScarbStarknetContractArtifact {
+    sierra: PathBuf,
+    casm: Option<PathBuf>,
+}
 
 #[derive(Args)]
 #[command(about = "Declare a contract to starknet", long_about = None)]
 pub struct Declare {
-    /// Path to the sierra compiled contract
-    #[clap(short = 's', long = "sierra-contract-path")]
-    pub sierra_contract_path: Utf8PathBuf,
-
-    /// Path to the casm compiled contract
-    #[clap(short = 'c', long = "casm-contract-path")]
-    pub casm_contract_path: Utf8PathBuf,
+    /// contract name
+    pub contract: String,
 
     /// Max fee for the transaction. If not provided, max fee will be automatically estimated
     #[clap(short, long)]
     pub max_fee: Option<u128>,
 }
 
 pub async fn declare(
-    sierra_contract_path: &Utf8PathBuf,
-    casm_contract_path: &Utf8PathBuf,
+    contract_name: &str,
     max_fee: Option<u128>,
     account: &mut SingleOwnerAccount<&JsonRpcClient<HttpTransport>, LocalWallet>,
 ) -> Result<DeclareTransactionResult> {
+    let contract_name: String = contract_name.to_string();
+    let command_result = Command::new("scarb")
+        .current_dir(std::env::current_dir().context("Failed to obtain current dir")?)
+        .arg("build")
+        .output()
+        .context("Failed to start building contracts with Scarb")?;
+    let result_code = command_result.status.code().context("failed to obtain status code from scarb build")?;
+    if result_code != 0 {
+        panic!("scarb build returned non-zero exit code: {}", result_code);
+    }
+
+    // TODO #2141 improve handling starknet artifacts
+    let current_dir = std::env::current_dir()
+                .expect("Failed to get current directory")
+                .join("target/dev");
```

Comment:
Consider using `scarb manifest-path` instead of current dir as base path. 

For instance, in project structured as following:
```
-> project
    -> src
        -> lib.cairo
    -> Scarb.toml
```
`scarb build` run should exit successfully, producing `project/target` subdirectory.

Reply:
Yes, this can be done but in such a case, we'd have to either pass this from the main or obtain it once again using scarb metadata (that uses `current_dir` anyway, so there's no difference after all). If you know how to solve this without using `current_dir` at some point, please let me know.

Reply:
There is a command `scarb manifest-path`, which offloads Scarb manifest search to Scarb.

Reply:
Discussed offline, deferring to https://github.com/software-mansion/protostar/issues/2154

Reply:
Can you give us tldr here or under issue? we (Marcin and I) did not participate in this discussion, and have no idea what was the result of it, where one of us will probably have to handle this